tests/e2e/kafkatopic_webhook.go

```diff
@@ -0,0 +1,318 @@
+	. "github.com/onsi/gomega"
+)
+
+// TODO(mihalexa): move to k8s.go
+const (
+	dryRunStrategyNone   string = "none"
+	dryRunStrategyClient string = "client"
+	dryRunStrategyServer string = "server"
+)
+
+// TODO(mihalexa): move to k8s.go
+// applyK8sResourceFromTemplateWithDryRun is copy of applyK8sResourceFromTemplate which calls a "--dry-run=<strategy>" kubectl command
+func applyK8sResourceFromTemplateWithDryRun(kubectlOptions k8s.KubectlOptions, templateFile string, values map[string]interface{}, dryRunStrategy string) (string, error) {
+	By(fmt.Sprintf("Generating K8s manifest from template %s for dry-run apply", templateFile))
+	var manifest bytes.Buffer
+	rawTemplate, err := os.ReadFile(templateFile)
+	if err != nil {
+		return "", err
+	}
+	t := template.Must(template.New("template").Funcs(sprig.TxtFuncMap()).Parse(string(rawTemplate)))
+	err = t.Execute(&manifest, values)
+	if err != nil {
+		return "", err
+	}
+
+	By("Replicating terratest's k8s.KubectlApplyFromStringE")
+	tmpfile, err := k8s.StoreConfigToTempFileE(GinkgoT(), manifest.String())
+	if err != nil {
+		return "", err
+	}
+	defer os.Remove(tmpfile)
+
+	return k8s.RunKubectlAndGetOutputE(GinkgoT(), &kubectlOptions, "apply", "-f", tmpfile, "--dry-run="+dryRunStrategy, "--output=yaml")
+}
+
+func testWebhooks() bool {
+	return When("Testing webhooks", func() {
+		// temporary section; to be refactored after kubeconfig injection PR
+		var kubectlOptions k8s.KubectlOptions
+		var err error
+		kubectlOptions, err = kubectlOptionsForCurrentContext()
+		if err != nil {
+			GinkgoT().Fail()
+		}
+
+		testWebhookKafkaTopic(kubectlOptions)
+	})
+}
+
+func testWebhookKafkaTopic(kubectlOptions k8s.KubectlOptions) {
+	kubectlOptions.Namespace = koperatorLocalHelmDescriptor.Namespace
+	requireCreatingKafkaCluster(kubectlOptions, "../../config/samples/simplekafkacluster.yaml")
+	testWebhookCreateKafkaTopic(kubectlOptions)
+	testWebhookUpdateKafkaTopic(kubectlOptions)
+	requireDeleteKafkaCluster(kubectlOptions, kafkaClusterName)
+}
+
+func testWebhookCreateKafkaTopic(kubectlOptions k8s.KubectlOptions) bool {
+	return When("Testing KafkaTopic Create", func() {
+		BeforeAll(func() {
+			Expect(isExistingK8SResource(kubectlOptions, kafkaKind, kafkaClusterName)).To(BeTrue())
+		})
+
+		It("Test non-existent KafkaCluster", func() {
+			output, err := applyK8sResourceFromTemplateWithDryRun(kubectlOptions,
+				kafkaTopicTemplate,
+				map[string]interface{}{
+					"Name":      testInternalTopicName,
+					"TopicName": testInternalTopicName,
+					"Namespace": kubectlOptions.Namespace,
+					"ClusterRef": map[string]string{
+						"Name":      kafkaClusterName + "NOT",
+						"Namespace": kubectlOptions.Namespace,
+					},
+				},
+				dryRunStrategyServer,
+			)
+			Expect(err).To(HaveOccurred())
+			// Example error: The KafkaTopic "topic-test-internal" is invalid: spec.clusterRef.name: Invalid value: "kafkaNOT": kafkaCluster 'kafkaNOT' in the namespace 'kafka' does not exist
```

Comment:
**Question**: Is this really meaningful with line 108 being present? I see it as somewhat of a noise, but could be wrong.

Reply:
In such cases my practice was to build the message clearly from 1 level of indirection if variables are defined - but mostly no variables for reuse in end to end tests for clarity purposes so string literals are the interface we are using.
I'm biased towards my experience but I can try and convince myself about this approach as well.